src/library_fs.js

```diff
@@ -403,29 +403,39 @@ FS.staticInit();` +
         FS.FSStream = /** @constructor */ function() {
           this.shared = { };
         };
-        FS.FSStream.prototype = {
+        FS.FSStream.prototype = {};
```

Comment:
Can you explain how / why this fixes the issue?

Reply:
Before we weren't creating getters and setters, we were creating a property called `position` whose value was an object with two attributes `get` and `set`. So `stream.position` wasn't returning a number it returned this object. Then when `stream.position = 0` is executed, it writes over the object. So streams weren't sharing position. And also the missing `:` below.

Reply:
But does this line specifically need to change or can you revert to `FS.FSStream.prototype = {` and avoid the extra `Object.defineProperties`?  (and all the extra annotations?)

Reply:
Oh I see `defineProperties` is actually needed setup the getters and setters correctly.

SGTM.

Does closure really need al those annotations ?  Seems like it should be able to infer those. but maybe not.

Reply:
Yeah without the annotations closure raises an error.